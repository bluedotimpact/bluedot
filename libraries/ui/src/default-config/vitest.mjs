/* eslint-disable import/no-extraneous-dependencies */
import { defineConfig, defaultExclude } from 'vitest/config';
import react from '@vitejs/plugin-react';
import * as dotenv from 'dotenv';

/**
 * @param {import('vitest/config').UserConfig} [config] 
 * @returns {import('vitest/config').UserConfig}
 */
export const withDefaultBlueDotVitestConfig = async (config) => defineConfig({
  ...config,
  plugins: [react(), ...(config?.plugins ?? [])],
  test: {
    environment: 'happy-dom',
    env: dotenv.config({ path: '.env.test' }).parsed,

    ...config?.test,

    exclude: [...defaultExclude, '**/.{turbo,next}/**', ...(config?.test?.exclude ?? [])],
  },
});
