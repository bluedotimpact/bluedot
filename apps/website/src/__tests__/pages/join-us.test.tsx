import { render } from '@testing-library/react';
import { describe, expect, test } from 'vitest';
import JoinUsPage from '../../pages/join-us';
import { TrpcWrapper } from '../trpcWrapper';

describe('JoinUsPage', () => {
  test('should render correctly', () => {
    const { container } = render(<JoinUsPage />, { wrapper: TrpcWrapper });
    expect(container).toMatchSnapshot();
  });
});
