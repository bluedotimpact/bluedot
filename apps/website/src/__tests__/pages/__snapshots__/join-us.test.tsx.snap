// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JoinUsPage > should render correctly 1`] = `
<div>
  <div>
    <div
      class="hero-section bg-bluedot-darker flex flex-row justify-center items-center w-full px-spacing-x"
    >
      <div
        class="hero-section__content max-w-[865px] py-12"
      >
        <div
          class="hero-section__mini-title text-color-text-on-dark text-center uppercase tracking-wider text-sm font-semibold mb-4"
        >
          Join Us
        </div>
        <h1
          class="hero-section__title text-color-text-on-dark text-center bluedot-h1"
        >
          Join us in our mission to ensure humanity safely navigates the transition to transformative AI.
        </h1>
        <div
          class="flex justify-center mt-8"
        >
          <a
            class="cta-button flex items-center justify-center rounded-sm transition-all duration-200 text-sm px-4 py-3 w-fit font-[650] whitespace-nowrap cursor-pointer not-prose cta-button--primary bg-bluedot-normal link-on-dark"
            href="#open-roles-anchor"
            tabindex="0"
          >
            <span
              class="cta-button__text"
            >
              See open roles
            </span>
            <span
              class="cta-button__chevron ml-3"
            >
              <svg
                class="cta-button__chevron-icon size-2"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 320 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M310.6 233.4c12.5 12.5 12.5 32.8 0 45.3l-192 192c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3L242.7 256 73.4 86.6c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0l192 192z"
                />
              </svg>
            </span>
          </a>
        </div>
      </div>
    </div>
    <div
      class="breadcrumbs bg-color-canvas border-b border-color-divider w-full py-space-between"
    >
      <nav
        aria-label="Breadcrumbs"
        class="breadcrumbs__nav section-base flex flex-row justify-between"
      >
        <ol
          class="breadcrumbs__list flex"
        >
          <li
            class="breadcrumbs__item flex items-center"
          >
            <a
              class="bluedot-a not-prose breadcrumbs__link no-underline"
              href="/"
              tabindex="0"
            >
              Home
            </a>
            <span
              class="breadcrumbs__separator mx-2"
            >
              &gt;
            </span>
          </li>
          <li
            class="breadcrumbs__item flex items-center"
          >
            <a
              class="bluedot-a not-prose breadcrumbs__link no-underline"
              href="/join-us"
              tabindex="0"
            >
              Join us
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <section
      class="section section-body"
    >
      <div
        class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
      >
        <div
          class="section-heading__content flex-1 flex flex-col gap-2"
        >
          <h2
            class="section-heading__title relative ml-[-0.08em] bluedot-h2"
          >
            Careers at BlueDot Impact
          </h2>
        </div>
      </div>
      <div
        class="section__body"
      >
        <div
          class="progress-dots flex justify-center space-x-2 my-6"
        >
          <span
            class="size-2 bg-bluedot-normal rounded-full animate-bounce"
            style="animation-delay: 0ms;"
          />
          <span
            class="size-2 bg-bluedot-normal rounded-full animate-bounce"
            style="animation-delay: 150ms;"
          />
          <span
            class="size-2 bg-bluedot-normal rounded-full animate-bounce"
            style="animation-delay: 300ms;"
          />
        </div>
      </div>
    </section>
    <section
      class="section section-body culture-section"
    >
      <div
        class="section__body"
      >
        <div
          class="culture-section__container flex lg:flex-row flex-col gap-space-between"
        >
          <div
            class="culture-section__text-container flex flex-col gap-5 max-w-[436px]"
          >
            <div
              class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
            >
              <div
                class="section-heading__content flex-1 flex flex-col gap-2"
              >
                <h2
                  class="section-heading__title relative ml-[-0.08em] bluedot-h2"
                >
                  Our culture
                </h2>
              </div>
            </div>
            <p
              class="bluedot-p not-prose"
            >
              At BlueDot Impact, our culture is defined by high agency, rapid experimentation, and direct feedback. We move fast and iterate quickly because AI capabilities are advancing rapidly and our work is time-sensitive.
            </p>
            <p
              class="bluedot-p not-prose"
            >
              But speed doesn't mean sacrificing support â€“ we invest heavily in each other's growth and wellbeing, creating an environment where ambitious, mission-driven people can do their best work.
            </p>
            <p
              class="bluedot-p not-prose"
            >
              We seek exceptional people who combine a sense of urgency with the empathy to help others thrive, whether that's fellow team members or the thousands of professionals in our global community.
            </p>
          </div>
          <div
            class="culture-section__image-grid grid mt-6 grid-cols-2 gap-x-2 gap-y-4 sm:mt-0 sm:grid-cols-3 sm:gap-4"
          >
            <img
              alt="BlueDot Impact team"
              class="culture-section__image max-h-[100px] sm:max-h-[178px] w-full object-cover rounded-sm"
              src="/images/culture/culture_dewi_v1.jpeg"
            />
            <img
              alt="BlueDot Impact team"
              class="culture-section__image max-h-[100px] sm:max-h-[178px] w-full object-cover rounded-sm"
              src="/images/culture/culture_talking_v2.jpeg"
            />
            <img
              alt="BlueDot Impact team"
              class="culture-section__image max-h-[100px] sm:max-h-[178px] w-full object-cover rounded-sm"
              src="/images/culture/culture_happy_v1.jpeg"
            />
            <img
              alt="BlueDot Impact team"
              class="culture-section__image max-h-[100px] sm:max-h-[178px] w-full object-cover rounded-sm"
              src="/images/culture/culture_will_v1.jpeg"
            />
            <img
              alt="BlueDot Impact team"
              class="culture-section__image max-h-[100px] sm:max-h-[178px] w-full object-cover rounded-sm"
              src="/images/culture/culture_prod-eng_v1.jpeg"
            />
            <img
              alt="BlueDot Impact team"
              class="culture-section__image max-h-[100px] sm:max-h-[178px] w-full object-cover rounded-sm"
              src="/images/culture/culture_keyboard_v1.jpeg"
            />
          </div>
        </div>
      </div>
    </section>
    <section
      class="section section-body values-section"
    >
      <div
        class="section-heading__title-container flex justify-between items-center gap-space-between mb-6"
      >
        <div
          class="section-heading__content flex-1 flex flex-col gap-2"
        >
          <h2
            class="section-heading__title relative ml-[-0.08em] bluedot-h2"
          >
            Our values
          </h2>
        </div>
      </div>
      <div
        class="section__body"
      >
        <div
          class="slide-list relative overflow-hidden flex flex-col gap-space-between -mx-spacing-x md:mx-0 values-section__values--desktop"
        >
          <div
            class="slide-list__gradient-wrapper relative flex-1"
          >
            <div
              class="slide-list__gradient-overlay absolute inset-0 pointer-events-none z-10 flex"
            >
              <div
                class="slide-list__gradient-left absolute inset-y-0 w-8 from-cream-normal to-transparent opacity-50 right-0 bg-linear-to-l"
              />
            </div>
            <div
              class="slide-list__slides flex flex-1 h-full items-stretch overflow-x-scroll gap-y-spacing-y md:gap-x-space-between scrollbar-hidden transition-transform duration-300 snap-x snap-mandatory"
              style="scroll-padding-inline: 0px;"
            >
              <div
                class="slide-list__slide shrink-0 snap-start pl-spacing-x md:pl-0 last:pr-spacing-x md:last:pr-0"
                data-width="calc((100% - var(--space-between) * 0) * (1 - 0.15))"
                style="scroll-margin-left: -0px;"
              >
                <div
                  class="card flex items-start transition-transform duration-200 flex-col values-section__value w-[350px]"
                >
                  <div
                    class="card__image-container w-full mb-4"
                  >
                    <img
                      alt="Own the mission"
                      class="card__image max-w-full max-h-full object-cover rounded-lg "
                      src="/images/values/mission.gif"
                    />
                  </div>
                  <div
                    class="card__content flex gap-6 w-full flex-1 justify-between flex-col"
                  >
                    <div
                      class="card__text"
                    >
                      <p
                        class="card__title bluedot-h4 mb-2"
                      >
                        Own the mission
                      </p>
                      <p
                        class="card__subtitle bluedot-p "
                      >
                        We prioritize mission success above all, take responsibility for outcomes, and commit extraordinary effort to achieve extraordinary goals.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="slide-list__slide shrink-0 snap-start pl-spacing-x md:pl-0 last:pr-spacing-x md:last:pr-0"
                data-width="calc((100% - var(--space-between) * 0) * (1 - 0.15))"
                style="scroll-margin-left: -0px;"
              >
                <div
                  class="card flex items-start transition-transform duration-200 flex-col values-section__value w-[350px]"
                >
                  <div
                    class="card__image-container w-full mb-4"
                  >
                    <img
                      alt="Find the fastest, best way to do everything"
                      class="card__image max-w-full max-h-full object-cover rounded-lg "
                      src="/images/values/fast.gif"
                    />
                  </div>
                  <div
                    class="card__content flex gap-6 w-full flex-1 justify-between flex-col"
                  >
                    <div
                      class="card__text"
                    >
                      <p
                        class="card__title bluedot-h4 mb-2"
                      >
                        Find the fastest, best way to do everything
                      </p>
                      <p
                        class="card__subtitle bluedot-p "
                      >
                        We're high agency: constantly growing our skills, finding creative solutions, and making the impossible possible.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="slide-list__slide shrink-0 snap-start pl-spacing-x md:pl-0 last:pr-spacing-x md:last:pr-0"
                data-width="calc((100% - var(--space-between) * 0) * (1 - 0.15))"
                style="scroll-margin-left: -0px;"
              >
                <div
                  class="card flex items-start transition-transform duration-200 flex-col values-section__value w-[350px]"
                >
                  <div
                    class="card__image-container w-full mb-4"
                  >
                    <img
                      alt="Say the uncomfortable truth"
                      class="card__image max-w-full max-h-full object-cover rounded-lg "
                      src="/images/values/speak.gif"
                    />
                  </div>
                  <div
                    class="card__content flex gap-6 w-full flex-1 justify-between flex-col"
                  >
                    <div
                      class="card__text"
                    >
                      <p
                        class="card__title bluedot-h4 mb-2"
                      >
                        Say the uncomfortable truth
                      </p>
                      <p
                        class="card__subtitle bluedot-p "
                      >
                        We speak essential truths with kindness: giving feedback, challenging ideas, and supporting each other's growth.
                      </p>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex justify-between items-center gap-space-between"
          >
            <button
              aria-label="Previous slide"
              class="slide-list__nav-button p-3 border rounded-lg transition-colors m-[3px] mt-0 border-charcoal-light text-black hover:bg-gray-50 hover:cursor-pointer active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-hidden focus:ring-2 focus:ring-bluedot-light"
              disabled=""
              type="button"
            >
              <svg
                class="slide-list__nav-icon size-5"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M15 19l-7-7 7-7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
            <div
              class="slide-list__progress w-full relative h-1"
            >
              <div
                class="slide-list__progress-track absolute h-px top-px w-full bg-color-divider"
              />
              <div
                class="slide-list__progress-track absolute h-full bg-bluedot-normal"
                style="width: 33.33333333333333%;"
              />
            </div>
            <button
              aria-label="Next slide"
              class="slide-list__nav-button p-3 border rounded-lg transition-colors m-[3px] mt-0 border-charcoal-light text-black hover:bg-gray-50 hover:cursor-pointer active:bg-gray-100 disabled:opacity-50 disabled:cursor-not-allowed focus:outline-hidden focus:ring-2 focus:ring-bluedot-light"
              type="button"
            >
              <svg
                class="slide-list__nav-icon size-5"
                fill="none"
                stroke="currentColor"
                viewBox="0 0 24 24"
              >
                <path
                  d="M9 5l7 7-7 7"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                />
              </svg>
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
